README.md

```diff
@@ -2,6 +2,113 @@
+{
+  "entrypoint: "index.html",
+  "builds": {
+    {
+      "name": "modern",
+      "browserCapabilities": ["push", "es2015"]
+    },
+    {
+      "name": "fallback",
+      "browserCapabilities": []
+    }
+  }
+}
+```
+
+### Capabilities
+
+The `browserCapabilities` field defines the browser features required for that build. prpl-server analyzes the request user-agent header and picks the best build for which all capabilities are met. If multiple builds are compatible, the one with more capabilities is preferred. If there is a tie, the build that comes earlier in the configuration file wins.
+
+You should always include a fallback build with no capability requirements. If you don't, prpl-server will warn at startup, and will return a 400 error on entrypoint requests to browsers for which no build can be served.
+
+## Entrypoint
+
+In the [PRPL pattern](https://developers.google.com/web/fundamentals/performance/prpl-pattern/), the *entrypoint* is a small HTML file that acts as the application bootstrap.
+
+prpl-server will serve the entrypoint from the best compatible build from `/`, and from any path that does not have a file extension and is not an existing file.
+
+prpl-server expects that each build subdirectory contains its own entrypoint file. By default it is `index.html`, or you can specify another name with the `entrypoint` configuration file setting.
+
+Note that because the entrypoint is served from many URLs, and varies by user-agent, cache hits for the entrypoint will be minimal, so it should be kept as small as possible.
+
+## Base paths
+
+Since prpl-server serves resources from build subdirectories, your application source can't know the absolute URLs of build-specific resources upfront.
+
+For most documents in your application, the solution is to use relative URLs to refer to other resources in the same build, and absolute URLs to refer to resources shared across builds. However, since the *entrypoint* is served from URLs that do not match its location in the build tree, relative URLs will not resolve correctly.
+
+The solution we recommend is to place a [base](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/base) tag in your entrypoint to anchor its relative URLs to the correct build subdirectory, regardless of the URL the entrypoint was served from. You may then use relative URLs to refer to build-specific resources from your entrypoint, as though you were in your build subdirectory. Put `<base href="/">` in your source entrypoint, so that URLs resolve when serving your source directly during development. In your build pipeline, update each entrypoint's base tag to match its build subdirectory (e.g. `<base href="/modern/">`).
+
+If you are using polymer-cli, set `{basePath: true}` on each build configuration to perform this base tag update automatically.
+
+Note that base tags only affect relative URLs, so to refer to resources shared across builds from your entrypoint, use absolute URLs as you normally would.
```

Comment:
"shared across builds" sounds vague to mel, maybe "at specific URLs on your domain" or "outside of your application's assets" or something, or both?